File: backend/Mizan.Domain/Entities/Recipe.cs
namespace Mizan.Domain.Entities;

public class Recipe
{
    public Guid Id { get; set; }
    public Guid? UserId { get; set; }
    public Guid? HouseholdId { get; set; }
    public string Title { get; set; } = string.Empty;
    public string? Description { get; set; }
    public int Servings { get; set; } = 1;
    public int? PrepTimeMinutes { get; set; }
    public int? CookTimeMinutes { get; set; }
    public string? SourceUrl { get; set; }
    public string? ImageUrl { get; set; }
    public bool IsPublic { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }

    // Navigation properties
    public virtual User? User { get; set; }
    public virtual Household? Household { get; set; }
    public virtual ICollection<RecipeIngredient> Ingredients { get; set; } = new List<RecipeIngredient>();
    public virtual ICollection<RecipeInstruction> Instructions { get; set; } = new List<RecipeInstruction>();
    public virtual RecipeNutrition? Nutrition { get; set; }
    public virtual ICollection<RecipeTag> Tags { get; set; } = new List<RecipeTag>();
    public virtual ICollection<MealPlanRecipe> MealPlanRecipes { get; set; } = new List<MealPlanRecipe>();
    public virtual ICollection<FoodDiaryEntry> DiaryEntries { get; set; } = new List<FoodDiaryEntry>();
}

File: frontend/types/recipe.ts
import Ingredient from "./ingredient";

// Recipe type matching the .NET API response
export default interface Recipe {
	id: string;
	title: string;
	description?: string;
	imageUrl?: string;
	servings: number;
	prepTimeMinutes?: number;
	cookTimeMinutes?: number;
	calories?: number;
	protein?: number;
	carbs?: number;
	fat?: number;
	fiber?: number;
	ingredients?: RecipeIngredient[];
	instructions?: string[];
	tags?: string[];
	creatorId?: string;
	isPublic: boolean;
	createdAt: string;
	updatedAt: string;
}

export interface RecipeIngredient {
	ingredientText: string;
	foodName?: string;
	amount?: number;
	unit?: string;
}

export interface RecipeInput {
	title: string;
	description?: string;
	imageUrl?: string;
	ingredients: {
		ingredient: Ingredient;
		amount: number;
		unit: string;
	}[];
	servings: number;
	instructions: string[];
	tags: string[];
}
